flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/ComponentMetricGroup.java

```diff
@@ -33,19 +36,42 @@
  * group could for example include the task attempt number (more fine grained identification), or
  * exclude it (for continuity of the namespace across failure and recovery).
  */
-public abstract class ComponentMetricGroup extends AbstractMetricGroup {
+public abstract class ComponentMetricGroup<P extends AbstractMetricGroup> extends AbstractMetricGroup<P> {
```

Comment:
The `JobMetricGroup` is annotated as `internal`. I was just wondering whether we should do the same for this metric group and the others (to which it applies) as well.